src/rust/bitbox02-rust/src/hww/api/bluetooth.rs

```diff
@@ -163,6 +163,7 @@ async fn process_upgrade(
 
     if response.is_ok() {
         hal.ui().status("Upgrade\nsuccessful", true).await;
+        bitbox02::reset_ble();
```

Comment:
This does not work so well in practice. I suspect it's because the app disconnects in this moment, the success response below is never delivered to the app, so the Rust async task is never dropped. When reconnecting, the app tries to send some API requests, and the device denies it because it thinks it's still running a task.

Disconnecting here also revealed a bug in the the Bluetooth swift code: if the connection breaks in when the app still expects (part of) a response, the readBlocking loop hangs forever waiting for the semaphore:

https://github.com/BitBoxSwiss/bitbox-wallet-app/blob/fa0349170d63752071c613638d3a874ece8cc3d8/frontends/ios/BitBoxApp/BitBoxApp/Bluetooth.swift#L311